File: app/src/main/java/com/example/testeandroidv2/adapter/Adapter.java
package com.example.testeandroidv2.adapter;

import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.TextView;

import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;

import com.example.testeandroidv2.R;
import com.example.testeandroidv2.model.Lancamento;

import java.util.List;

public class Adapter extends RecyclerView.Adapter<Adapter.MyViewHolder> {

    private List<Lancamento> listaLancamentos;

    public Adapter(List<Lancamento> lista) {
        this.listaLancamentos = (List<Lancamento>) lista;
    }


    @NonNull
    @Override
    public MyViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {

        View itemLista = LayoutInflater.from(parent.getContext()).inflate(R.layout.adapter_lancamento, parent, false);
        return new MyViewHolder(itemLista);
    }

    @Override
    public void onBindViewHolder(@NonNull MyViewHolder holder, int position) {

        Lancamento lancamento = listaLancamentos.get(position);
        holder.categoria.setText(lancamento.getCategoria());
        holder.conta.setText(lancamento.getConta());
        holder.data.setText(lancamento.getData());
        holder.valor.setText(lancamento.getValor());

    }

    @Override
    public int getItemCount() {
        return listaLancamentos.size();
    }

    public class MyViewHolder extends RecyclerView.ViewHolder{

        TextView categoria;
        TextView conta;
        TextView data;
        TextView valor;

        public MyViewHolder(@NonNull View itemView) {
            super(itemView);

            categoria = itemView.findViewById(R.id.textCategoria);
            conta = itemView.findViewById(R.id.textConta);
            data = itemView.findViewById(R.id.textData);
            valor = itemView.findViewById(R.id.textValor);

        }
    }

}

File: app/src/main/java/com/example/accentureprojectbank/Adapter.kt
package com.example.accentureprojectbank

import android.content.Intent
import android.view.LayoutInflater
import android.view.ViewGroup
import androidx.recyclerview.widget.RecyclerView

class Adapter(val listaInf: MutableList<Informacoes>) : RecyclerView.Adapter<ViewHolder>() {

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): ViewHolder {
        val view = LayoutInflater.from(parent.context).inflate(R.layout.extrato_view_holder, parent, false)
        return ViewHolder(view)
    }

    override fun getItemCount(): Int = listaInf.size


    override fun onBindViewHolder(holder: ViewHolder, position: Int) {

        val payment = holder.pagamento
        payment.text = listaInf[position].titulo

        val account = holder.conta
        account.text = listaInf[position].descricao

        val date = holder.data
        date.text = listaInf[position].data

        val value = holder.valor
        value.text = listaInf[position].valor.toString()
    }
}

